src/allmydata/test/cli/test_invite.py

```diff
@@ -1,75 +1,74 @@
 """
-Ported to Pythn 3.
+Tests for ``tahoe invite``.
 """
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
 
-from future.utils import PY2
-if PY2:
-    from future.builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, list, object, range, str, max, min  # noqa: F401
-
-import os
-import mock
 import json
+import os
+from functools import partial
 from os.path import join
+from typing import Awaitable, Callable, Optional, Sequence, TypeVar
 
-try:
-    from typing import Optional, Sequence
-except ImportError:
-    pass
-
-from twisted.trial import unittest
 from twisted.internet import defer
+from twisted.trial import unittest
+
+from ...client import read_config
+from ...scripts import runner
+from ...util.jsonbytes import dumps_bytes
 from ..common_util import run_cli
 from ..no_network import GridTestMixin
 from .common import CLITestMixin
-from ...client import (
-    read_config,
-)
-
-class _FakeWormhole(object):
-
-    def __init__(self, outgoing_messages):
-        self.messages = []
-        for o in outgoing_messages:
-            assert isinstance(o, bytes)
-        self._outgoing = outgoing_messages
-
-    def get_code(self):
-        return defer.succeed(u"6-alarmist-tuba")
-
-    def set_code(self, code):
-        self._code = code
-
-    def get_welcome(self):
-        return defer.succeed(
-            {
-                u"welcome": {},
-            }
-        )
-
-    def allocate_code(self):
-        return None
-
-    def send_message(self, msg):
-        assert isinstance(msg, bytes)
-        self.messages.append(msg)
-
-    def get_message(self):
-        return defer.succeed(self._outgoing.pop(0))
-
-    def close(self):
-        return defer.succeed(None)
-
-
-def _create_fake_wormhole(outgoing_messages):
-    outgoing_messages = [
-        m.encode("utf-8") if isinstance(m, str) else m
-        for m in outgoing_messages
-    ]
-    return _FakeWormhole(outgoing_messages)
+from .wormholetesting import IWormhole, MemoryWormholeServer, memory_server
+
+
+async def open_wormhole() -> tuple[Callable, IWormhole, str]:
```

Comment:
Should this maybe be in `wormholetesting.py`, or is it super-specific to this module?

Reply:
Hmm yea, dunno.  It's kind of more specific than the other stuff in `wormholetesting.py` since it does that `run_cli` stuff.

I guess I'm inclined to leave it for now.  Maybe when there's more wormhole functionality in Tahoe that needs testing, those requirements will suggest a better answer.